src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -127,6 +159,34 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
     this
   }
 
+  /*
+   * Implemented by means of a buffer to keep the last n elements from being returned during iteration.
+   */
+  def dropRight(n: Int): Iterator[A] = {
```

Comment:
I wouldn't ever want to do two traversals. We should either do buffering or not have the operations at all (like in the 2.12 collections). If we go with buffering I'd opt for a lazy implementation that doesn't buffer before the first call to `hasNext` or `next` of the resulting iterator.